index.js

```diff
@@ -15,6 +15,52 @@ const plugin_options = new Set([
 	'preprocess'
 ]);
 
+const parsePkg = function(dir) {
+	const pkgFile = path.join(dir, 'package.json');
+
+	try {
+		return JSON.parse(fs.readFileSync(pkgFile, 'utf-8'));
+	} catch (e) {
+		return false;
+	}
+}
+
+const findPkg = function(name, importer) {
+	let dir, pkg;
+
+	try {
+		const file = `${name}/package.json`;
+		const resolved = relative.resolve(file, path.dirname(importer));
+		dir = path.dirname(resolved);
+		pkg = require(resolved);
+	} catch (err) {
+		if (err.code === 'MODULE_NOT_FOUND') return {pkg: null, dir};
+		if (err.code === 'ERR_PACKAGE_PATH_NOT_EXPORTED') {
+			dir = path.dirname(relative.resolve(name));
```

Comment:
```suggestion
			dir = path.dirname(relative.resolve(name, path.dirname(importer));
```

I think we need to specify the importer too. Perhaps we can abstract this as a function since it's used in the try block too.